internal/client/tanzukubernetescluster/nodepool/nodepool_resource.go

```diff
@@ -0,0 +1,103 @@
+
+import (
+	"net/url"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/transport"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	tkcmodels "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/tanzukubernetescluster"
+	tkcnodepool "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/tanzukubernetescluster/nodepool"
+)
+
+const (
+	apiVersionAndGroup      = "/v1alpha1/managementclusters"
+	apiNodepoolsPath        = "nodepools"
+	provisioners            = "provisioners"
+	queryParamKeySortBy     = "sortBy"
+	tanzukubernetesclusters = "tanzukubernetesclusters"
+)
+
+// New creates a new tanzu kubernetes cluster node pool resource service API client.
+func New(transport *transport.Client) ClientService {
+	return &Client{Client: transport}
+}
+
+/*
+Client for cluster node pool resource service API.
+*/
+type Client struct {
+	*transport.Client
+}
+
+// ClientService is the interface for Client methods.
+type ClientService interface {
+	TanzuKubernetesNodePoolResourceServiceGet(fn *tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolFullName) (*tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIResponse, error)
+
+	TanzuKubernetesNodePoolResourceServiceCreate(request *tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIRequest) (*tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIResponse, error)
+
+	TanzuKubernetesNodePoolResourceServiceList(cluster *tkcmodels.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterFullName) (*tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolListNodepoolsResponse, error)
+
+	TanzuKubernetesNodePoolResourceServiceDelete(fn *tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolFullName) error
+
+	TanzuKubernetesNodePoolResourceServiceUpdate(request *tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIRequest) (*tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIResponse, error)
+}
+
+// TanzuKubernetesNodePoolResourceServiceGet implements ClientService.
+func (c *Client) TanzuKubernetesNodePoolResourceServiceGet(fn *tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolFullName) (*tkcnodepool.VmwareTanzuManageV1alpha1ManagementclusterProvisionerTanzukubernetesclusterNodepoolAPIResponse, error) {
+	queryParams := url.Values{}
+
+	requestURL := helper.ConstructRequestURL(apiVersionAndGroup, fn.ManagementClusterName, provisioners, fn.ProvisionerName, tanzukubernetesclusters, fn.TanzuKubernetesClusterName, apiNodepoolsPath, fn.Name).AppendQueryParams(queryParams).String()
```

Comment:
You can use something like [this](https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/244/files#diff-88df8fde84060e6da833d73e57df8068fa3bb108790536324044c837702de930R23) to construct the request URL?